### DISCUSSION TITLE
Error when syncing to OneDrive for Business Shared Folder

### DISCUSSION THREAD

**Discussion by liljerri** [abraunegg/onedrive#1711] [Q&A] [answered]
I can sync with my OneDrive for Business folder, but I cannot sync with shared folders. I get the following error. This is a new issue, I was able to sync yesterday without issue. I did make any configuration changes so I don't know what would be causing the new error. 

Any help would be much appreciated!

I get the following when running `onedrive --synchronize --sync-shared-folders`
```
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Syncing this OneDrive Business Shared Folder: FOLDER
Retry sync count: 1: Key not found: remaining
Syncing changes from OneDrive ...
Syncing this OneDrive Business Shared Folder: FOLDER
Retry sync count: 2: Key not found: remaining
Syncing changes from OneDrive ...
Syncing this OneDrive Business Shared Folder: FOLDER
Giving up on sync after three attempts: Key not found: remaining
std.json.JSONException@std/json.d(627): Key not found: remaining

??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException).enforce!(inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x7fe3f8e8edb6]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(immutable(char)[]) [0x7fe3f8ee8636]
??:? [0x55c233aed72a]
??:? [0x55c233aed206]
??:? [0x55c233b372e0]
??:? [0x55c233b2f63e]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7fe3f8bfcb9b]
??:? _d_run_main2 [0x7fe3f8bfc9b7]
??:? _d_run_main [0x7fe3f8bfc80d]
??:? [0x7fe3f882efcf]
??:? __libc_start_main [0x7fe3f882f07c]
??:? [0x55c233ad876d]
std.net.curl.CurlException@std/net/curl.d(4402): A libcurl function was given a bad argument on handle null

??:? pure @safe bool std.exception.enforce!(std.net.curl.CurlException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x7fe3f8e8f05c]
??:? void std.net.curl.HTTP.clearRequestHeaders() [0x7fe3f8ef614f]
??:? [0x55c233b294af]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7fe3f8bfcb9b]
??:? _d_run_main2 [0x7fe3f8bfc9b7]
??:? _d_run_main [0x7fe3f8bfc80d]
??:? [0x7fe3f882efcf]
??:? __libc_start_main [0x7fe3f882f07c]
??:? [0x55c233ad876d]
```

**Comment by abraunegg** [comment]
@liljerri 
This appears to be an API change.

Please submit a new bug report following the support process.

**Reply by liljerri** [reply]
Looks like you already merged pr1713 into main, do you still want me to submit a bug report? I'm happy to if you like

**Comment by abraunegg** [comment, marked as answer]
@liljerri 
Please can you test the following PR to see if this resolves this issue:
```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/1713/head:pr1713
git checkout pr1713
./configure; make clean; make;
```
To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any options needed>
```

When running the PR, your version should be: `onedrive v2.4.13-30-g4bfae61` or greater

**Reply by liljerri** [reply]
It works! That was super fast, thank you!

**Reply by cbs310** [reply]
Though the fix worked for me indeed (I had the exact same problem), I now get bitten by:
```  
onedrive --synchronize --upload-only --confdir=~/.config/onedrive-zzz-shared
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
ERROR: OneDrive quota information is being restricted. Please fix by speaking to your OneDrive / Office 365 Administrator.
...
Uploading new items of ~/Cloud/zzz-shared
Skipping item './xxx/yyy.txt' due to insufficient free space available on OneDrive
```
This message about 'the quota information being restricted' has occurred as far as I remember; however it never stopped onedrive from uploading the files. Could the patch have changed the behaviour relating to the quota information?

**Reply by abraunegg** [reply]
@cbs310 
~~Please raise a new bug following the support process please.~~

**Reply by abraunegg** [reply]
@cbs310 
Please rebuild your client from 'master' please and see if this issue is now resolved for you.

**Reply by cbs310** [reply]
Indeed, it disappeared!?! Great, thank you!

**Reply by cbs310** [reply]
For completeness: The original "ERROR: OneDrive quota information is being restricted. Please fix by speaking to your OneDrive / Office 365 Administrator." remains, but behaviour is back to the old behaviour, where uploads are working as they used to.

**Reply by abraunegg** [reply]
@cbs310 
> The original `ERROR: OneDrive quota information is being restricted. Please fix by speaking to your OneDrive / Office 365 Administrator.` remains

Yes .. because your OneDrive / Azure AD Security Policy deployed by your organisation is restricting the 'quota' details from being exposed by the OneDrive API ..... this is something I cannot fix - only your Azure AD team can fix.
